### PR TITLE
database seeded, api implemented

### PR DESCRIPTION
**Pull request by 3-Hz**

### CODE REVIEW

File: db/model.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
+const db = require('./db.js');
+
+
+class Model {
+
+  addItem(name, brand, avg_rating, num_ratings) {
+    let values = [name, brand, avg_rating, num_ratings];
+    let queryString = `
+      INSERT INTO items (name, brand, avg_rating, num_ratings)
+      VALUES (?, ?, ?, ?)
```
Comment (bschaaf1017) [inline db/model.js:10]: Im assuming the ?'s are just place holders for now?

File: db/seed.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+const generateItems = (numItems) => {
+  let items = [];
+  for (let item_id = 0; item_id < NUM_ITEMS; item_id++) {
+    let item = {
+      item_id: item_id + 1,
+      name: faker.commerce.productName(),
+      brand: faker.company.companyName(),
+      avg_rating: rI(25, 50),
```
Comment (bschaaf1017) [inline db/seed.js:20]: just a question what is rI / what does it do?

File: .gitignore [hunk 3]
```diff
@@ -22,5 +22,4 @@ coverage
```
Comment (bschaaf1017) [inline .gitignore:22]: shoot i totally forgot to set up git ignore ill have to do that!!

File: db/db.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+const Promise = require('bluebird');
+
+const connection = MySql.createConnection({
+  host: 'localhost',
+  user: 'student',
+  password: 'student'
+})
+
+const db = Promise.promisifyAll(connection, { multiArgs: true });
+
```
Comment (bschaaf1017) [inline db/db.js:11]: i need to figure this out it seems like its keeping things alot smoother for you unlike mine (promisifyAll)

**Comment by 3-Hz** [comment]
`db/data/photos.js` was used to generate `db/data/photos.json`, which is read-only data, so they can be ignored. The real meat and bones of this feature lie in the `db/model.js`, `db/seed.js`, and `server/server.js` files.

**Comment by 3-Hz** [comment]
I think I can separate out the model method invocations into a separate `controller` file to clean up the `server.js` file a bit as well.

File: server/server.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+app.get('/api/items/:item_id/photos', (req, res) => {
+  m.getPhotos(req.params.item_id)
+  .then(qR => res.json(qR[0]))
+  .catch(err => {
+    res.status(404).end();
+    console.log(err);
+  });
+})
+
+
```
Comment (bschaaf1017) [inline server/server.js:45]: everything looks awesome man! gave me a lot of ideas for my own code so it was really hard for me to find anything critical to say..

**Review by bschaaf1017 — APPROVED** [review]

File: db/model.js (hunk 1 above)
Comment (3-Hz) [reply db/model.js:10]: They are wildcards that allow you to use values from the second parameter of `db.query()`

File: db/seed.js (hunk 2 above)
Comment (3-Hz) [reply db/seed.js:20]: `rI(min, max)` is a helper function that I defined at the top of the file. It generates a random integer from `min` (inclusive) to `max` (inclusive).

**Pull request merged**